src/main/java/com/google/devtools/build/lib/bazel/bzlmod/SingleExtensionEvalFunction.java

```diff
@@ -207,21 +217,18 @@ public SkyValue compute(SkyKey skyKey, Environment env)
                         extensionId, nonVisibleRepoNames)));
       }
     }
-    if (lockfileMode.equals(LockfileMode.ERROR)) {
-      boolean extensionShouldHaveBeenLocked =
-          moduleExtensionMetadata.map(metadata -> !metadata.getReproducible()).orElse(true);
-      // If this extension was not found in the lockfile, and after evaluation we found that it is
-      // not reproducible, then error indicating that it was expected to be in the lockfile.
-      if (lockedExtension == null && extensionShouldHaveBeenLocked) {
-        throw new SingleExtensionEvalFunctionException(
-            ExternalDepsException.withMessage(
-                Code.BAD_LOCKFILE,
-                "The module extension '%s'%s does not exist in the lockfile",
-                extensionId,
-                extension.getEvalFactors().isEmpty()
-                    ? ""
-                    : " for platform " + extension.getEvalFactors()));
-      }
+    if (lockfileMode.equals(LockfileMode.ERROR)
+        && !moduleExtensionMetadata.map(ModuleExtensionMetadata::getReproducible).orElse(false)) {
+      // The extension is not reproducible and can't be in the lockfile, since an existing (but
```

Comment:
to confirm, the `lockedExtension == null` branch of the condition was _never_ necessary, right? Even before this PR, `lockedExtension != null` would mean `tryGettingValueFromLockFile` either returned non-null or threw an exception.

Reply:
Yes, that's true in ERROR mode.